### DISCUSSION TITLE
Ground-Truth Errors Large, but Model Predicts Well

### DISCUSSION THREAD

**Discussion by mollywestbrook** [talmolab/sleap#1518] [Help!]
Hello all! I've been finding some issue with our current model's error metrics (this was referenced awhile ago in issue #1392, but was never quite solved. Our model is up to 1010 frames, and is performing quite well. Screengrab of tracks from a group of videos I batch-tracked last night.
![Slide1](https://github.com/talmolab/sleap/assets/48096260/f5f7a519-f925-4e2e-869e-03653e83b033)
Haven't done a full error analysis on these videos yet, but the initial visual check seems pretty much exactly where we want the model to be. Not perfect, but definitely workable. In the last iteration of the model I was seeing most node MLE above 0.9, which seems quite good, and based on these results I expect the same from this model. However, error rate of our training validation is still extraordinarily high:
![Slide2](https://github.com/talmolab/sleap/assets/48096260/6e4a44ee-e44d-4312-af31-be62f53da37b)
I went back and checked the training validation frames from our most recent training session, and the predictions are absurdly off the animals. A few examples:
![image](https://github.com/talmolab/sleap/assets/48096260/c89776c4-bccb-48c3-b864-e3bd59b53006)
![image](https://github.com/talmolab/sleap/assets/48096260/aaeb50c5-7f2b-4ffc-9668-c95e74d277fd)
![image](https://github.com/talmolab/sleap/assets/48096260/5c2bf74d-e048-4728-acc3-d7b646345c08)
I don't understand why our model is maintaining such a high error rate during validation, but then predicts on videos quite well. Any ideas? Happy to provide more info.

**Comment by roomrys** [comment]
Hi @mollywestbrook,

I have an inkling that this shifting of predictions likely has to do with the `resize_and_pad_to_target` and related parameters in the training_config.json. This parameter should be set to `true`, but what I believe might be happening is that the `target_height` and `target_width` are set to the incorrect dimensions. I see you have variable sized videos... so, what I would try is:

---
1. Find the training_config.json that you have been selecting for use in the GUI
2. Find the `resize_and_pad_to_target`, `target_height`, and `target_width` parameters
3. Ensure that `target_height` and `target_width` are both set to "null" (no quotes)
4. Save the training_config.json
5. Repeat 1-4 for each training_config.json to be used (2 for the top-down)
---
6. (Re)Open the Training Widget GUI
7. Select the edited training_config.json
8. Optionally, if on 1.3.3+ "Copy to Clipboard" and paste the config info somewhere to verify your changes were applied
9. Repeat 6-8 for each training_config.json to be used
---
10. Rerun training
---

Let me know if you need more guidance/clarification.

Thanks,
Liezl

**Reply by mollywestbrook** [reply]
Hi @roomrys,
``
Sorry for the delay in getting back, but I did go to verify these parameters and indeed,  `resize_and_pad_to_target` is set to `true`, and `target_height` and `target_width` are both set to `null`. Is there any other possibility for why this error may be happening?

**Comment by roomrys** [comment]
No solution yet, but as an update:

We reviewed this issue in our meeting today and it looks like the cause is most likely a temporal shift or just a general frame retrieval error (predicting on a different frame than it is assigning those predictions to). Seeing as this only happens on the validation set... I wonder if this might be related to the training/validation split function and possibly [retaining select hidden config parameters when using the GUI](https://github.com/talmolab/sleap/pull/1053) (without [setting the validation indices back to their defaults](https://github.com/talmolab/sleap/pull/1331)).
